### ISSUE TITLE
Failed to load dynamic library with flutter_test

### ISSUE THREAD

**Issue by dossiman** [realm/realm-dart#968]
### What happened?

When running tests with flutter_test on macos, the following error is thrown:
``Failed to load dynamic library 'librealm_dart.dylib'``

The workaround was to install realm_dart to pubspec.yaml and run ``dart run realm_dart install``

Now the error becomes:
``Failed to load dynamic library 'realm.framework/realm'``

The workaround was to edit ``_getAppDirectory`` in ``realm_core.dart`` and replace
``if (!isFlutterPlatform) {``
with
``if (!isFlutterPlatform || Platform.environment.containsKey('FLUTTER_TEST')) {``

I don't know if this is the best practice but can something like this be added so we can run flutter test with realm on the command line?

### Repro steps

run ``flutter test`` on a test file that import realm

### Version

3.3.4

### What Realm SDK flavor are you using?

Local Database only

### What type of application is this?

Flutter Application

### Client OS and version

macOS 12.6 21G115 darwin-arm

### Code snippets

import 'package:realm/realm.dart';

void main() {
  Realm.logger.info('is flutter? $isFlutterPlatform');
  Realm(Configuration.inMemory([]));
}

### Relevant log output

```shell
Invalid argument(s): Failed to load dynamic library 'realm.framework/realm': dlopen(realm.framework/realm, 0x0001): tried: '/Users/xxx/.fvm/versions/stable/bin/cache/artifacts/engine/darwin-x64/./realm.framework/realm' (no such file), '/Users/xxx/.fvm/versions/stable/bin/cache/artifacts/engine/darwin-x64/../../../realm.framework/realm' (no such file), '/Users/xxx/.fvm/versions/stable/bin/cache/artifacts/engine/darwin-x64/Frameworks/realm.framework/realm' (no such file), '/Users/xxx/.fvm/versions/stable/bin/cache/artifacts/engine/darwin-x64/./realm.framework/realm' (no such file), '/Users/xxx/.fvm/versions/stable/bin/cache/artifacts/engine/darwin-x64/../../../realm.framework/realm' (no such file), '/Users/xxx/.fvm/versions/stable/bin/cache/artifacts/engine/darwin-x64/Frameworks/realm.framework/realm' (no such file), 'realm.framework/realm' (no such file), '/System/Library/Frameworks/realm.framework/realm' (no such file)
```

**Comment by nielsenko** [comment]
Hi @dossiman 

Thank you for trying out our beta. The issue you face is already known (#791, #870) and there is a PR (#929) with a fix that didn't get reviewed in time for 0.5.0+beta. I expect it to be merged before the next release.

**Comment by dossiman** [comment]
waiting for the fix in #929
